op-proposer/drivers/l2output/driver.go

```diff
@@ -42,11 +42,26 @@ type Config struct {
 	// L2OOAddr is the L1 contract address of the L2 Output Oracle.
 	L2OOAddr common.Address
 
-	// ChainID is the L1 chain ID used for proposal transaction signing
-	ChainID *big.Int
+	// From is the address to send transactions from
+	From common.Address
 
-	// Privkey used for proposal transaction signing
-	PrivKey *ecdsa.PrivateKey
+	// SignerFn is the function used to sign transactions
+	SignerFn bind.SignerFn
+}
+
+func SignerFnForPrivateKey(key *ecdsa.PrivateKey, chainID *big.Int) bind.SignerFn {
```

Comment:
this can probably live elsewhere... suggestions?

Reply:
I'd put it in `op-service`.

Reply:
ack, done in #4415 (i'm guessing op-service is versioned separately, let me know if this workflow is incorrect)